hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/OnlineDsAssignmentStrategy.java

```diff
@@ -106,4 +167,49 @@ JpaActionStatus createActionStatus(final JpaAction action, final String actionMe
         return result;
     }
 
+    private DistributionSetAssignmentResult sendDistributionSetAssignedEvent(
+            final DistributionSetAssignmentResult assignmentResult) {
+        final List<Action> filteredActions = assignmentResult.getActions().stream().filter(action -> {
+            final Status actionStatus = action.getStatus();
+            return !hasPendingCancellations(action.getTarget()) && Status.CANCELING != actionStatus
```

Comment:
Looks like almost the same as in line [110](https://github.com/eclipse/hawkbit/pull/833/files#diff-7913dcdd27955f4245f6c3d42eac8c9fR110)